sympy/physics/wigner.py

```diff
@@ -533,16 +536,56 @@ def wigner_9j(j_1, j_2, j_3, j_4, j_5, j_6, j_7, j_8, j_9, prec=None):
     for finite precision arithmetic and only useful for a computer
     algebra system [Rasch03]_.
     """
-    imin = 0
-    imax = min(j_1 + j_9, j_2 + j_6, j_4 + j_8)
 
+    imax = min(j_1 + j_9, j_2 + j_6, j_4 + j_8)
     sumres = 0
-    for kk in range(imin, int(imax) + 1):
-        sumres = sumres + (2 * kk + 1) * \
-            racah(j_1, j_2, j_9, j_6, j_3, kk, prec) * \
-            racah(j_4, j_6, j_8, j_2, j_5, kk, prec) * \
-            racah(j_1, j_4, j_9, j_8, j_7, kk, prec)
+    for kk in range(0, 2*(int(imax) + 1)):
+        k = S.Half*kk
+        try:
+            j_sum = j_1+j_2+j_9+j_6+j_4+j_8
+            sumres +=(-1)**(kk+2*j_sum)*(kk + 1) * \
+            racah(j_1, j_2, j_9, j_6, j_3, k, prec) * \
+            racah(j_4, j_6, j_8, j_2, j_5, k, prec) * \
+            racah(j_1, j_4, j_9, j_8, j_7, k, prec)
+        except:
```

Comment:
Bare excepts should not be used. It is only ValueError that is excepted and silently passed. If something else would turn up, that would be an error that should not be handled here.

Reply:
Thank you for explaining - changed to ValueError.